sdk/core/Azure.Core/tests/TestFramework/RecordedTestBase.cs

```diff
@@ -17,6 +17,13 @@ public abstract class RecordedTestBase : ClientTestBase
 
         public RecordedTestMode Mode { get; }
 
+        /// <summary>
+        /// Flag you can (temporarily) enable to save failed test recordings
+        /// and debug/re-run at the point of failure without re-running
+        /// potentially lengthy live tests.  This should never be checked in.
+        /// </summary>
+        public bool DEBUG_ONLY_SaveRecordingsOnFailure { get; set; } = false;
```

Comment:
Should we instead flow them into the TestLogger?

Reply:
That wouldn't save them though, right?  I want them saved so I can do a `dotnet msbuild /t:UpdateSessionRecords` and keep re-running the test while I figure out why the damn thing won't parse.

A very common scenario with this PR was letting a test run for a minute plus and return successfully from the service only to fail on the final parse in some uniquely odd way.  Being able to replay them even if the test didn't pass was super useful.  If we really don't want to check this in, I can just add it back manually whenever I need it.

Reply:
What about `#if DEBUG` around this property to make it truly debug only?

Reply:
That's a good suggestion.  I'll make the name slightly less visually gruesome then since the CI will catch this for us if we slip.

Reply:
Changed